azure-mgmt-keyvault/src/main/java/com/microsoft/azure/management/keyvault/implementation/VaultsImpl.java

```diff
@@ -91,10 +110,250 @@ protected VaultImpl wrapModel(VaultInner vaultInner) {
                 vaultInner,
                 super.manager(),
                 graphRbacManager);
+    }    
+	
+	@Override
+	public PagedList<DeletedVault> listDeleted() {
+	    PagedList<DeletedVaultInner> listDeleted = this.inner().listDeleted();
+	    PagedListConverter<DeletedVaultInner, DeletedVault> converter =  
+	            new PagedListConverter<DeletedVaultInner, DeletedVault> () {
+	                @Override
+                    public Observable<DeletedVault> typeConvertAsync(DeletedVaultInner inner) {
+                        return Observable.just((DeletedVault) new DeletedVaultImpl(inner));
```

Comment:
@tiffanyachen the cast is not needed here (if DeletedVaultImpl class implements DeletedVault interface).